packages/preferences/src/browser/preferences-widget.ts

```diff
@@ -0,0 +1,258 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, named } from "inversify";
+import { Message } from "@phosphor/messaging";
+import { PreferencesMenuFactory } from "./preference-menu-factory";
+import { PreferencesDecorator } from "./preferences-decorator";
+import {
+    ContextMenuRenderer,
+    ExpandableTreeNode,
+    PreferenceSchemaProvider,
+    SelectableTreeNode,
+    PreferenceProperty,
+    PreferenceScope,
+    PreferenceService,
+    TreeModel,
+    TreeNode,
+    TreeProps,
+    TreeWidget,
+    WidgetManager,
+    PreferenceProvider,
+    Saveable
+} from "@theia/core/lib/browser";
+import {
+    USER_PREFERENCES_TREE_WIDGET_ID,
+    WORKSPACE_PREFERENCES_TREE_WIDGET_ID
+} from "./preference-frontend-contribution";
+import { SplitPanel } from "@phosphor/widgets/lib/splitpanel";
+import { UserPreferenceProvider } from "./user-preference-provider";
+import { WorkspacePreferenceProvider } from "./workspace-preference-provider";
+import { EditorWidget } from "@theia/editor/lib/browser";
+import { Emitter, Event, MaybePromise } from '@theia/core';
+import { Widget } from "@phosphor/widgets";
+
+export abstract class PreferencesContainer extends SplitPanel implements Saveable {
```

Comment:
If the widget relies on another `Saveable` then it should implement `SaveableSource` instead.

Reply:
In this case `savable` is retrieved immediately after widget creation, but editor widget is initialized asynchronously.